Added Installation section which includes basic instructions on how t…

Arsivnet opened the pull request:
Hello. I was checking the repo and decided to try out. I added an "installation" part on the readme that also specifies the standard.

The repo compiles successfully both for GNU99 and GNU23. I used both cmake and AutoTools. I modified CmakeLists.txt like so in line 32:
`SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu23")`
Same for GNU99.

When i tried strictly for C99 like this:
`SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99 -pedantic")`

It worked with AutoTools.
`./configure
make
sudo make install`

But C99 did NOT work when i tried cmake.
`mkdir build
cmake ..
make`

Gave an error like this:
`/home/user/qlibc/src/internal/qinternal.h:80:52: error: ‘PTHREAD_MUTEX_RECURSIVE’ undeclared (first use in this function); did you mean ‘PTHREAD_MUTEX_RECURSIVE_NP’?
   80 |             pthread_mutexattr_settype(&_mutexattr, PTHREAD_MUTEX_RECURSIVE); \`
   
  
fixes #115

Arsivnet committed:
Added Installation section which includes basic instructions on how to install and also specifies that it needs at least GNU99 but also compiles at GNU23

wolkykim commented on README.md:
````diff
@@ -12,6 +12,25 @@ APIs with a consistent API look.
+sudo make install
+```
+
+Alternatively, you can use Cmake. qLibc requires at least GNU99 and confirmed to be compiling in GNU23
````
I expect all the Linux/Linux-like users to use autotools and CMake for Windows.
Do you think we should include Windows installation instructions instead?

Arsivnet replied:
I think both would be good. There's always new people coming to Linux these days. For windows we could add steps both for visual studio and VScode, also maybe a cmd/powershell option.

wolkykim commented on README.md:
````diff
@@ -12,6 +12,25 @@ APIs with a consistent API look.
+cd build
+cmake ..
+sudo make install
+```
+
````
Let's take this part out for now. Not sure if this should be the officially recommended way.

Arsivnet replied:
Just removed that section and pushed

Arsivnet committed:
Update README.md

Removed Cmake alternative build method from the installation section as it's not certain if it should be an official recommended way

wolkykim reviewed (approved):
LGTM
